src/routes/admin/ciraconfig/ciraValidator.ts

```diff
@@ -110,9 +137,28 @@ export const ciraUpdateValidator = (): any => {
       .matches('^[a-zA-Z0-9$@$!%*#?&-_~^]+$')
       .withMessage('MPS user name accepts letters, numbers, special characters and no spaces'),
     check('password')
+      .if((value, { req }) => req.body.generateRandomPassword === false)
       .optional()
       .matches('^(?=.*[0-9])(?=.*[!@#$%^&*])(?=.*[a-z])(?=.*[A-Z])[a-zA-Z0-9$@$!%*#?&-_~^]{8,32}$')
       .withMessage('Password should contain at least one lowercase letter, one uppercase letter, one numeric digit,and one special character and password length should be in between 8 to 32.'),
+    check('passwordLength')
+      .if((value, { req }) => req.body.generateRandomPassword === true)
+      .optional()
+      .isInt({ min: 8, max: 32 })
```

Comment:
I believe max length is 16 for the mps password

Reply:
done